lib/upgrade_lts_contract.py

```diff
@@ -35,13 +35,19 @@
     "18.04": "bionic",
     "20.04": "focal",
     "20.10": "groovy",
+    "21.04": "hirsute",
+    "21.10": "impish",
+    "22.04": "jammy",
 }
 
 current_codename_to_past_codename = {
     "xenial": "trusty",
     "bionic": "xenial",
     "focal": "bionic",
     "groovy": "focal",
+    "hirsute": "groovy",
+    "impish": "hirsute",
+    "jammy": "hirsute",
```

Comment:
@lucasmoura 

I think this is used to infer the release before the `do-release-upgrade`, and my understanding is that the current/near-future upgrade paths are:
* trusty -> xenial
* xenial -> bionic
* bionic -> focal
* focal -> impish
* impish -> jammy
* focal -> jammy

So I would expect the changes in this PR to look like this:
```suggestion
    "impish": "focal",
    "jammy": "impish",
    "jammy": "focal",
```

except of course we can't have two entries for jammy. Is there a way we can detect what the prior release was instead of inferring based on the current release?

Reply:
That's a good point. Since this runs at the end of `do-release-upgrade`, I don't think we have the original release info anymore. However, since the main idea is to apply any contract deltas between the releases, I think we can assume the `focal -> jammy` situation here, since no services are available for impish. What do you think ?

Reply:
Yeah that's probably fine, and I suppose our `behave-upgrade-21.10` test should catch any errors that that assumption might create. So yes I think that makes sense, let's try it

Reply:
@orndorffgrant updated. I have also removed from the list the releases we no longer support as mentioned by @renanrodrigo